static/js/reactions.js

```diff
@@ -142,15 +142,19 @@ exports.reaction_navigate = function (e, event_name) {
     var first_emoji = get_emoji_at_index(0);
     var selected_emoji = get_selected_emoji();
     var selected_index = find_index_for_emoji(selected_emoji);
-
+    var emoji_popover_filter = $('.emoji-popover-filter')[0];
     // special cases
     if (event_name === 'down_arrow') {
         if ($('.emoji-popover-filter').is(':focus') && first_emoji) { // move down into emoji map
             $(first_emoji).focus();
         }
     } else if (event_name === 'up_arrow') {
         if (selected_emoji && selected_index < 6) { // move up into reaction filter
-            $('.emoji-popover-filter').focus();
+            emoji_popover_filter.focus();
+            setTimeout(function () {
+                emoji_popover_filter.selectionStart = emoji_popover_filter.value.length;
+                emoji_popover_filter.selectionEnd =  emoji_popover_filter.value.length;
+            }, 0);
```

Comment:
Secondly, the setTimeout - since the events are handled by the Web api and callback queues. The focus gets executed sometime later (i.e when the runtime stack becomes empty). So, in order to make sure that I set the cursor position after the element has been focused, I have to pass it as a callback to setTimeout. Which gets executed in similar fashion to focus event (i.e its added to the queue next or after focus) . So, the cusor position is set after focus. But this creates milisecond delay making the flicker noticeable.